src/stronghold/mod.rs

```diff
@@ -217,17 +227,16 @@ impl StrongholdAdapter {
         }
 
         // Try to read a snapshot to check if the password is correct
-        if self.snapshot_path.is_some() {
-            let result = self.read_stronghold_snapshot().await;
-            if let Err(err) = result {
-                // TODO: replace with actual error matching when updated to the new Stronghold version
-                if let crate::Error::StrongholdProcedureError(ref err_msg) = err {
-                    if !err_msg.contains("IOError") {
-                        // If loading the snapshot failed but wasn't an IOError, then the password was incorrect and we
-                        // clear it
-                        *self.key.lock().await = None;
-                        return Err(err);
-                    }
+        let result = self.read_stronghold_snapshot().await;
+        if let Err(err) = result {
+            // TODO: replace with actual error matching when updated to the new Stronghold version
+            // TODO: but the error is not an enum
+            if let crate::Error::StrongholdProcedureError(ref err_msg) = err {
+                if !err_msg.to_string().contains("IOError") {
+                    // If loading the snapshot failed but wasn't an IOError, then the password was incorrect and we
+                    // clear it
+                    *self.key.lock().await = None;
```

Comment:
`EncryptionKey` is already wrapped in `Zeroize` which is good, maybe better to explicitly call `zeroize()` to make sure the data is cleared out immediately.